paddle/fluid/framework/ir/generate_pass.cc

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/generate_pass.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+void InitGeneratePattern(PDPattern* pattern, const proto::PassDesc& pass_desc) {
+  const proto::BlockDesc& block = pass_desc.pattern().blocks(0);
```

Comment:
If we don't support cross block pass, we should mention it in python/c++ interface or API comment.

Reply:
Usually, a subgraph is only in one block. We can support corss block in future if necessary.
I will add descriptions in next PR.